.github/workflows/build-test.yml

```diff
@@ -7,15 +7,16 @@ on:
 jobs:
   build:
     name: Test Builds
-    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest, macOS-latest]
+        go-version: [1.18.x, 1.19.x]
+        os: [ubuntu-latest, windows-latest, macOS-12]
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Set up Go
         uses: actions/setup-go@v3
         with:
-          go-version: 1.18
```

Comment:
We can enable caching using  `cache: true` it will cache build outputs and dependencies .
It will save build time x 6

Reply:
in the cases of other workflows, that's helpful, but in the case of the build test, we actually wanted to re-run everything to ensure we are capturing all the cases and avoided using `cache` here.

> https://github.com/mvdan/github-actions-golang/issues/22#issuecomment-871259749

let me know if that makes sense.

Reply:
Yes , that makes sense we can adopt caching in other workflows where goal is not unit testing .

Reply:
maybe at https://github.com/projectdiscovery/nuclei/blob/master/.github/workflows/template-validate.yml 

Here we are using actions@cache but cache-key is `Linux-go` (which is not correct and hence cache doesnot expire  which defeats purpose of caching)  .  actions@setup-go actually uses `actions@cache` underneath but properly configures keys etc .